### ISSUE TITLE
Add DEPS-based package cherry pick instructions to cherry pick procedure

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/sdk#52232]
During the 3.0 release it came up on the analyzer team a few times that it would be good to have a solid procedure for cherry picking changes on our DEPS dependencies in the SDK.  The analyzer team has three important ones:  `dart-lang/tools`, `dart-lang/dartdoc`, and `dart-lang/linter`.   We managed to get by for 3.0 by either accepting the risk of pulling in a fair number of unrelated changes (linter) or avoiding landing big changes during the beta period (dartdoc), but there was always the possibility that the technical risk would be too high and we'd block a critical change because of this.

Having a written procedure to follow would make it easier to validate our procedures for doing such a cherry pick and not accidentally introduce more risk.

**Comment by athomas** [comment]
@sigurdm Did you end up writing this up after we did this last time?

**Comment by sigurdm** [comment]
Yes! It is the last section of https://github.com/dart-lang/sdk/wiki/Cherry-picks-to-a-release-channel.

@athomas Actually I wanted to ask you to read through it, and see if it makes sense (the wiki doesn't seem to have a review flow).

**Comment by athomas** [comment]
Neat! I might do some minor changes but this captures the essence of it.

Direct link for posterity:
https://github.com/dart-lang/sdk/wiki/Cherry-picks-to-a-release-channel#cherry-picking-a-commit-in-a-dependency
